package batchtank_messages;

//option optimize_for = LITE_RUNTIME;

enum Sensor {
    HEATERSENSOR = 0;
    COOLERSENSOR = 1;
    INLETSENSOR = 2;
    OUTLETSENSOR = 3;
}

enum Output {
    HEATER = 0;
    COOLER = 1;
    INLETPUMP = 2;
    OUTLETPUMP = 3;
}

message BaseMessage {
    repeated Sample sample = 1;
    repeated ControlSignal signal = 2;
    optional Register register = 3;
    optional Sensor getSensor = 4;
    optional bool endConnection = 5;
}

// Used for registering for a period time etc. 
message Register {
    repeated Sensor type = 1;
    // period in millis.
    required uint64 periodTime = 2;
}

message ControlSignal {
    required double value = 1;
    required Output type = 2;
}

message Sample {
    required double value = 1;
    required Sensor type = 2;
}

